File: app/src/main/java/be/digitalia/fosdem/model/RoomStatus.java
package be.digitalia.fosdem.model;

import androidx.annotation.ColorRes;
import androidx.annotation.StringRes;
import be.digitalia.fosdem.R;

public enum RoomStatus {
	OPEN(R.string.room_status_open, R.color.room_status_open),
	FULL(R.string.room_status_full, R.color.room_status_full),
	EMERGENCY_EVACUATION(R.string.room_status_emergency_evacuation, R.color.room_status_emergency_evacuation);

	private final int nameResId;
	private final int colorResId;

	RoomStatus(@StringRes int nameResId, @ColorRes int colorResId) {
		this.nameResId = nameResId;
		this.colorResId = colorResId;
	}

	@StringRes
	public int getNameResId() {
		return nameResId;
	}

	@ColorRes
	public int getColorResId() {
		return colorResId;
	}
}

File: app/src/main/kotlin/be/digitalia/fosdem/model/RoomStatus.kt
package be.digitalia.fosdem.model

import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import be.digitalia.fosdem.R

enum class RoomStatus(
    @param:StringRes @get:StringRes val nameResId: Int,
    @param:ColorRes @get:ColorRes val colorResId: Int,
    @param:DrawableRes @get:DrawableRes val iconResId: Int
) {
    OPEN(R.string.room_status_open, R.color.room_status_open, 0),
    FULL(R.string.room_status_full, R.color.room_status_full, R.drawable.ic_warning_white_18sp),
    EMERGENCY_EVACUATION(
        R.string.room_status_emergency_evacuation,
        R.color.room_status_emergency_evacuation,
        R.drawable.ic_warning_white_18sp
    )
}